src/opendr/perception/pose_estimation/hr_pose_estimation/high_resolution_learner.py

```diff
@@ -0,0 +1,815 @@
+                                                                  max_width, stride, upsample_ratio)
+
+            total_keypoints_num = 0
+            all_keypoints_by_type = []
+            for kpt_idx in range(18):
+                total_keypoints_num += extract_keypoints(avg_heatmaps[:, :, kpt_idx], all_keypoints_by_type,
+                                                         total_keypoints_num)
+
+            pose_entries, all_keypoints = group_keypoints(all_keypoints_by_type, avg_pafs)
+
+            for kpt_id in range(all_keypoints.shape[0]):
+                all_keypoints[kpt_id, 0] = (all_keypoints[kpt_id, 0] * stride / upsample_ratio - pad[1]) / scale
+                all_keypoints[kpt_id, 1] = (all_keypoints[kpt_id, 1] * stride / upsample_ratio - pad[0]) / scale
+
+            for i in range(all_keypoints.shape[0]):
+                for j in range(all_keypoints.shape[1]):
+                    if j == 0:  # Adjust offset if needed for evaluation on our HR datasets
+                        all_keypoints[i][j] = round((all_keypoints[i][j] + xmin) - offset)
+                    if j == 1:  # Adjust offset if needed for evaluation on our HR datasets
+                        all_keypoints[i][j] = round((all_keypoints[i][j] + ymin) - offset)
+
+            current_poses = []
+            for n in range(len(pose_entries)):
+                if len(pose_entries[n]) == 0:
+                    continue
+                pose_keypoints = np.ones((num_keypoints, 2), dtype=np.int32) * -1
+                for kpt_id in range(num_keypoints):
+                    if pose_entries[n][kpt_id] != -1.0:  # keypoint was found
+                        pose_keypoints[kpt_id, 0] = int(all_keypoints[int(pose_entries[n][kpt_id]), 0])
+                        pose_keypoints[kpt_id, 1] = int(all_keypoints[int(pose_entries[n][kpt_id]), 1])
+                pose = Pose(pose_keypoints, pose_entries[n][18])
+                current_poses.append(pose)
+
+        return current_poses
+
+    def init_model(self):
+        if self.model is None:
+            # No model loaded, initializing new
+            if self.backbone == "mobilenet":
+                self.model = PoseEstimationWithMobileNet(self.num_refinement_stages, use_stride=self.use_stride)
+            elif self.backbone == "mobilenetv2":
+                self.model = PoseEstimationWithMobileNetV2(self.num_refinement_stages,
+                                                           width_mult=self.mobilenetv2_width)
+            elif self.backbone == "shufflenet":
+                self.model = PoseEstimationWithShuffleNet(self.num_refinement_stages,
+                                                          groups=self.shufflenet_groups)
+        else:
+            raise UserWarning("Tried to initialize model while model is already initialized.")
+        self.model.to(self.device)
+
+    def load(self, path, verbose=False):
```

Comment:
This function should be documented. Also, we should explain that this tool works with models trained with LightweightOpenPose.